ansible/roles/operator-pipeline/tasks/main.yml

```diff
@@ -49,25 +50,29 @@
         operator-pipeline.key: "{{ lookup('file', operator_pipeline_private_key_local_path, rstrip=False) | b64encode }}"
         operator-pipeline.pem: "{{ lookup('file', operator_pipeline_private_cert_local_path, rstrip=False) | b64encode }}"
 
-
-- name: Create Operator pipeline quay token secret
-  no_log: yes
-  k8s:
-    state: present
-    force: yes
-    namespace: "{{ oc_namespace }}"
-    definition:
-      apiVersion: v1
-      kind: Secret
-      type: kubernetes.io/dockerconfigjson
-      metadata:
-        name: operator-pipeline-registry-token-{{ suffix }}
-        labels:
-          app: operator-pipeline
-          suffix: "{{ suffix }}"
-          env: "{{ env }}"
-      data:
-        .dockerconfigjson: "{{ lookup('file', operator_pipeline_quay_token, rstrip=False) | b64encode }}"
+- name: Grant registry access
+  block:
+    - name: Create Operator pipeline quay token secret
+      no_log: yes
+      k8s:
+        state: present
+        force: yes
+        namespace: "{{ oc_namespace }}"
+        definition:
+          apiVersion: v1
+          kind: Secret
+          type: kubernetes.io/dockerconfigjson
+          metadata:
+            name: operator-pipeline-registry-token-{{ suffix }}
+            labels:
+              app: operator-pipeline
+              suffix: "{{ suffix }}"
+              env: "{{ env }}"
+          data:
+            .dockerconfigjson: "{{ lookup('file', operator_pipeline_quay_token, rstrip=False) | b64encode }}"
+    - name: Link secret with pipeline service account (for imagestream operations)
+      ansible.builtin.shell: |
```

Comment:
Generally using a shell in ansible is not recommended because it is not idempotent. Is there a way how define it using k8s module?

Reply:
I didn't found any :(

Reply:
If not, I am also fine with keeping it as is.